src/backend/services/logger/utils.py

```diff
@@ -20,3 +22,11 @@ def get_logger(self) -> BaseLogger:
         else:
             # Default to StructuredLogging
             return StructuredLogging(level, renderer)
+
+
+def log_and_raise_http_exception(logger: BaseLogger, error_message: str):
```

Comment:
Nit: can we move this method to the logger class? This way we could call the function like `logger.log_and_raise(message)`

Reply:
done!